plugin/buycourses/src/buy_course_plugin.class.php

```diff
@@ -3566,6 +3629,33 @@ public function getServiceSaleCouponCode($serviceSaleId)
         return $couponCode['code'];
     }
 
+    /**
+     * @return array
+     */
+    public function getCecabankSignature($saleReference, $price)
+    {
+        $urlOk = api_get_path(WEB_PLUGIN_PATH).'buycourses/src/cecabank_success.php';
+        $urlKo = api_get_path(WEB_PLUGIN_PATH).'buycourses/src/cecabank_cancel.php';
+
+        $cecabankParams = $this->getCecabankParams();
+        $signature = $cecabankParams['crypto_key']
+        .$cecabankParams['merchant_id']
+        .$cecabankParams['acquirer_bin']
+        .$cecabankParams['terminal_id']
+        .$saleReference
+        .$price * 100
```

Comment:
This is not super important, because it is used relatively safely below, but given the fact that this value comes direct from `$_POST` without any type of validation, I would at least cast `$price` to (float), because not having a float there could trigger an error and reveal debug data if the server is not properly configured.
Actually... *checking* that it's a float or an integer would be a good idea too, in the spirit of avoiding errors.
Using PHP's feature, you could just enforce this by giving a type to the method parameters.

Reply:
Yeah, the $price must be a float so I going to cast it to that type. Also I´m going to change getCecabankSignature to enforce the float type for the method parameter